### PR TITLE
Dpp 902 temporary uln fix

### PR DESCRIPTION
**Pull request by davehowlett**
two new specs added to cover story.

### CODE REVIEW

File: src/PeriodEnd/CoInvestedPayments/SFA.DAS.Payments.Calc.CoInvestedPayments.IntegrationTests/FinishOnTime/WhenRefundPaymentsAreDue.cs [hunk 1]
```diff
@@ -89,24 +91,61 @@ public void ThenCoInvestedRefundPaymentsAreMadeWhenPartiallyRefunding()
...
+            TestDataHelper.AddPaymentDueForProvider(temporaryCommitmentId, _ukprn, amountDue: -1500, uln: newUln, learnerRefNumber:learnerRef);
+
+            // Act
+            _uut.Execute(_taskContext);
+
+            //Assert
+            var payments = TestDataHelper.GetPaymentsForCommitment(temporaryCommitmentId);
+            Assert.AreEqual(1, payments.Length);
+
+            //Assert.IsNotNull(payments.SingleOrDefault(p => p.FundingSource == (int)FundingSource.CoInvestedSfa && p.Amount == -1350));
```
Comment (tom-gough) [inline src/PeriodEnd/CoInvestedPayments/SFA.DAS.Payments.Calc.CoInvestedPayments.IntegrationTests/FinishOnTime/WhenRefundPaymentsAreDue.cs]: comment to be deleted

File: src/PeriodEnd/CoInvestedPayments/SFA.DAS.Payments.Calc.CoInvestedPayments.IntegrationTests/FinishOnTime/WhenNoPaymentsAreDue/WhenNoPaymentsAreDue.cs [hunk 2]
```diff
@@ -15,6 +15,9 @@ public class WhenNoPaymentsAreDue
         private CoInvestedPaymentsTask _uut;
         private IntegrationTaskContext _taskContext;
 
+        /// <summary>
+        /// Arranges this instance.
+        /// </summary>
```
Comment (tom-gough) [inline src/PeriodEnd/CoInvestedPayments/SFA.DAS.Payments.Calc.CoInvestedPayments.IntegrationTests/FinishOnTime/WhenNoPaymentsAreDue/WhenNoPaymentsAreDue.cs]: comment to be deleted

**Review by chrisfoster76 — APPROVED** [review]

**Comment by davehowlett** [comment]
closed in error

File: src/PeriodEnd/PaymentsDue/SFA.DAS.ProviderPayments.Calc.PaymentsDue.UnitTests/PaymentsDueProcessor/WhenChangingAUlnForALearner.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                    { ContextPropertyKeys.TransientDatabaseConnectionString, "" },
+                    { ContextPropertyKeys.LogLevel, "DEBUG" },
+                    { PaymentsContextPropertyKeys.YearOfCollection, "1718" }
+                });
+
+            _processor = new PaymentsDue.PaymentsDueProcessor(_logger, _mediator.Object, new ContextWrapper(_externalContext.Object));
+        }
+
+        [Test]
+        public void ThenPaymentsAreStillMadeCorrectly()
```
Comment (patelmaulik) [inline src/PeriodEnd/PaymentsDue/SFA.DAS.ProviderPayments.Calc.PaymentsDue.UnitTests/PaymentsDueProcessor/WhenChangingAUlnForALearner.cs:52]: could we change the name of the test to say, "then no payments are due" or something like that ?

**Comment by satya-sanikommu** [comment]
@davehowlett  Please merge Develop into this and resolve all the conflicts

**Pull request closed without merging**